src/CityOnPlanet.cpp

```diff
@@ -202,123 +312,345 @@ CityOnPlanet::~CityOnPlanet()
 CityOnPlanet::CityOnPlanet(Planet *planet, SpaceStation *station, const Uint32 seed)
 {
 	// beware, these are not used in this function, but are used in subroutines!
+	m_body = planet->GetSystemBody();
 	m_planet = planet;
 	m_frame = planet->GetFrame();
 	m_detailLevel = Pi::detail.cities;
 
-	m_buildings.clear();
-	m_buildings.reserve(DEFAULT_NUM_BUILDINGS);
+	m_rand.seed(seed);
 
-	const Aabb &aabb = station->GetAabb();
-	const matrix4x4d &m = station->GetOrient();
-	const vector3d p = station->GetPosition();
+	if (s_cityFlavours.empty()) {
+		// initialize needed variables to render nothing
+		AddStaticGeomsToCollisionSpace();
```

Comment:
This crashes if you remove `buildings/default.json`
At least because `m_cityType` points nowhere yet
And it is not clear what variables are initialized there, if we have everything 0.

Reply:
This should be fixed now, I hope. I'm not going to invest too much effort into it, as the case of having no building definitions at all usually means the game can't find the data directory and that causes a crash/error much earlier.